### PR TITLE
Allow passing volumes into mesh-task

### PR DESCRIPTION
**Pull request by pglass**
## Changes proposed in this PR:
- Add a `volumes` variable to mesh-task so that users can define additional volumes for the ECS task

## How I've tested this PR:
Unit test included

## How I expect reviewers to test this PR:
👀 

## Checklist:
- [x] Tests added
- [ ] CHANGELOG entry added 

    [HashiCorp engineers only. Community PRs should not add a changelog entry.]::
    [Changelog entries should use present tense, e.g. "Add support for..."]::

### CODE REVIEW

File: modules/mesh-task/main.tf [hunk 1]
```diff
@@ -207,6 +207,70 @@ resource "aws_ecs_task_definition" "this" {
     name = local.consul_binary_volume_name
   }
 
+  dynamic "volume" {
```
Comment (pglass) [inline modules/mesh-task/main.tf:210]: If there is a better way to do this, let me know.

It's translating a list of maps (`volumes = [{...}]`) into several blocks (`volume {}`). From what I can tell, Terraform doesn't support passing blocks into modules. And, it doesn't support dynamic names for the keys in these dynamic blocks.

Comment (erichaberkorn) [reply modules/mesh-task/main.tf:210]: Eek, this doesn't look fun. Did this come from somewhere else? If so, it would be great to link to the source.

**Review by erichaberkorn — APPROVED** [review]

File: modules/mesh-task/main.tf (hunk 1 above)
Comment (pglass) [reply modules/mesh-task/main.tf:210]: > Did this come from somewhere else?

No, I crafted this myself. I followed the [Terraform docs on dynamic blocks](https://www.terraform.io/docs/language/expressions/dynamic-blocks.html) and referred to the [volumes field in the `aws_ecs_task_definition`](https://registry.terraform.io/providers/hashicorp/aws/latest/docs/resources/ecs_task_definition#volume) and looked at a some examples of nested dynamic blocks.

(One way we maybe might avoid this is to implement mesh-task as a Terraform provider, but that's essentially a rewrite that we'd need to think through.)

**Review by ishustava — APPROVED** [review]
Sorry for the delay!

**Pull request merged**